sktime/contrib/hivecote/frequency_based/tsf.py

```diff
@@ -0,0 +1,212 @@
+
+    def predict(self, X):
+        """
+        Find predictions for all cases in X. Built on top of predict_proba
+        Parameters
+        ----------
+        X : The training input samples.  array-like or sparse matrix of shape = [n_samps, num_atts] or a data frame.
+        If a Pandas data frame is passed, the column _dim_to_use is extracted
+
+        Returns
+        -------
+        output : 1D array of predictions,
+        """
+
+        proba=self.predict_proba(X)
+        return [self.classes_[np.argmax(prob)] for prob in proba]
+
+    def predict_proba(self, X):
+        """
+        Find probability estimates for each class for all cases in X.
+        Parameters
+        ----------
+        X : array-like or sparse matrix of shape = [n_samps, num_atts]
+            The training input samples.  If a Pandas data frame is passed, the column _dim_to_use is extracted
+
+        Local variables
+        ----------
+        n_samps     : number of cases to classify
+        num_atts    : number of attributes in X, must match _num_atts determined in fit
+
+        Returns
+        -------
+        output : 2D array of probabilities,
+        """
+        if isinstance(X, pd.DataFrame):
+            if isinstance(X.iloc[0,self._dim_to_use],pd.Series):
+                X = np.asarray([a.values for a in X.iloc[:,0]])
+            else:
+                raise TypeError("Input should either be a 2d numpy array, or a pandas dataframe containing Series objects")
+
+        n_samps, num_atts = X.shape
+        if num_atts != self._num_atts:
+            raise TypeError(" ERROR number of attributes in the train does not match that in the test data")
+        sums = np.zeros((X.shape[0],self._num_classes), dtype=np.float64)
+        for i in range(0, self._num_trees):
+            transformed_x = np.empty(shape=(3*self._num_intervals,n_samps),dtype=np.float32)
+            for j in range(0,self._num_intervals):
+                means = np.mean(X[:, self._intervals[i][j][0]:self._intervals[i][j][1]], axis=1)
+                std_dev = np.std(X[:, self._intervals[i][j][0]:self._intervals[i][j][1]], axis=1)
+                slope = self.lsq_fit(X[:, self._intervals[i][j][0]:self._intervals[i][j][1]])
+                transformed_x[3*j]=means
+                transformed_x[3*j+1]=std_dev
+                transformed_x[3*j+2]=slope
+            transformed_x=transformed_x.T
+            sums += self._classifiers[i].predict_proba(transformed_x)
+
+        output = sums / (np.ones(self._num_classes) * self.n_estimators)
+        return output
+
+    def lsq_fit(self, Y):
+        """Find probability estimates for each class for all cases in X.
+        Parameters
+        ----------
+        Y: series to find slope of
+        """
+        x = np.arange(Y.shape[1]) + 1
+        slope = (np.mean(x * Y, axis=1) - np.mean(x) * np.mean(Y, axis=1)) / ((x * x).mean() - x.mean() ** 2)
+        return slope
+
+
+
+
+if __name__ == "__main__":
```

Comment:
That could be a unit test, instead of using `load_from_tsfile_to_dataframe` you could use `load_gunpoint`.